src/generators/legacy-json/types.d.ts

```diff
@@ -0,0 +1,83 @@
+import { ListItem } from 'mdast';
+
+export interface HierarchizedEntry extends ApiDocMetadataEntry {
+  hierarchyChildren: ApiDocMetadataEntry[];
+}
+
+export type Section =
+  | SignatureSection
+  | PropertySection
+  | EventSection
+  | MiscSection;
+
+export interface Meta {
+  changes: ApiDocMetadataChange[];
+  added?: string[];
+  napiVersion?: string[];
+  deprecated?: string[];
+  removed?: string[];
+}
+
+export interface SectionBase {
+  type: string;
+  name: string;
+  textRaw: string;
+  displayName?: string;
+  desc: string;
+  shortDesc?: string;
+  stability?: number;
+  stabilityText?: string;
+  meta?: Meta;
+}
+
+export interface ModuleSection extends SectionBase {
+  type: 'module';
+  source: string;
+  miscs?: MiscSection[];
+  modules?: ModuleSection[];
+  classes?: SignatureSection[];
+  methods?: MethodSignature[];
+  properties?: PropertySection[];
+  globals?: ModuleSection | { type: 'global' };
+  signatures?: SignatureSection[];
+}
+
+export interface SignatureSection extends SectionBase {
+  type: 'class' | 'ctor' | 'classMethod' | 'method';
+  signatures: MethodSignature[];
+}
+
+export interface Parameter {
+  name: string;
+  optional?: boolean;
+  default?: string;
+}
+
+export interface MethodSignature {
+  params: Parameter[];
+  return?: string;
+}
+
+export interface PropertySection extends SectionBase {
+  type: 'property';
+  [key: string]: string | undefined;
+}
+
+export interface EventSection extends SectionBase {
+  type: 'event';
+  params: ListItem[];
```

Comment:
I think we use `Array<>` notation everywhere else 🤔

Reply:
in any case on the website we only use Array<T>.